File: Py/01_helloworld.py
####################################################################################
# Simple way run script
print("Hello World")

####################################################################################
# Proper way to run script (inspired by C++)
# No code execution when importing functions
# Python use indentation and new lines
def main():
    print("Hello World 2")

if __name__ == "__main__":
    main()
    
####################################################################################
# Pass arguments to main
import sys

def main(argv):
    print(f"Hello {argv[1]}")

if __name__ == "__main__":
    main(sys.argv)

File: Cpp/01_helloworld.cpp
#include <iostream>

// ####################################################################################
// Simple way run script
int main(){
    std::cout << "Hello World" << std::endl;
    return 0;
}

// ####################################################################################
// Pass arguments to main
int main(int argc, char *argv[]){
    std::cout << "Hello " << argv[1] << std::endl;
    return 0;
}